### PR TITLE
Add dtype support to EmbeddingCenterContextBatchify

### PR DESCRIPTION
**Pull request by leezu**
## Description ##
This adds support for arbitrary dtype to EmbeddingCenterContextBatchify. Also some refactoring of the `data.py` in the word embedding scripts folder is done. Further `train_fasttext.py` is renamed to `train_sg_cbow.py` to better reflect it's scope.

### Essentials ###
- [x] Changes are complete (i.e. I finished coding on this PR)
- [X] All changes have test coverage
- [X] Code is well-documented

### Changes ###
- [X] Support arbitrary dtypes in EmbeddingCenterContextBatchify

## Comments ##
As NDArray does not support arbitrary dtype (e.g. string / python object), the returned types of EmbeddingCenterContextBatchify are changed to numpy arrays.
Todo: Update example notebook for changed `data.py`

### CODE REVIEW

**Comment by mli** [comment]
Job PR-416/8 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-416/8/index.html

**Comment by mli** [comment]
Job PR-416/11 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-416/11/index.html

**Comment by sxjscience** [comment]
A minor comment about the doc. LGTM.

File: scripts/word_embeddings/index.rst [hunk 1]
```diff
@@ -38,7 +38,7 @@ Word Embedding Training
 Besides loading pre-trained embeddings, the Gluon NLP toolkit also makes it easy
 to train embeddings.
 
-The following code block shows how to use Gluon NLP to train fastText or Word2Vec
+The following code block shows how to use Gluon NLP to train a SkipGram or CBOW
 models. The script and parts of the Gluon NLP library support just-in-time
```
Comment (sxjscience) [inline scripts/word_embeddings/index.rst]: Here, we should emphasize that we support both Word2Vec and fastText.

**Comment by mli** [comment]
Job PR-416/12 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-416/12/index.html

**Pull request merged**